peer/network.go

```diff
@@ -366,21 +371,21 @@ func (n *network) AppRequest(ctx context.Context, nodeID ids.NodeID, requestID u
 // Error returned by this function is expected to be treated as fatal by the engine
 // If [requestID] is not known, this function will emit a log and return a nil error.
 // If the response handler returns an error it is propagated as a fatal error.
-func (n *network) AppResponse(_ context.Context, nodeID ids.NodeID, requestID uint32, response []byte) error {
+func (n *network) AppResponse(ctx context.Context, nodeID ids.NodeID, requestID uint32, response []byte) error {
 	n.lock.Lock()
-	defer n.lock.Unlock()
 
 	if n.closed.Get() {
+		n.lock.Unlock()
 		return nil
```

Comment:
Are we okay with dropping responses (and timeouts) that should have been sent to the SDK router after we close the network?

Reply:
If we are not okay with this - then I think we'll need to be fairly careful around what gets passed through to the router.

Reply:
We chatted offline but agreed it was cleanest to leave the `Router` code as-is, and just drop the `closed` check on the server-end because we're guaranteed that the outstanding request is empty on shutdown because we empty it on `Shutdown` and stop sending requests after the flag is set.